app/lib/events/event.rb

```diff
@@ -1,21 +1,45 @@
 module Events
   class Event
+    EVENT_TYPES = %w[web_request].freeze
+
     def initialize
       @event_hash = {
         environment: HostingEnvironment.environment_name,
-        timestamp: Time.zone.now.iso8601,
+        occurred_at: Time.zone.now.iso8601,
       }
     end
 
     def as_json
       @event_hash.as_json
     end
 
+    def with_type(type)
+      raise 'Invalid analytics event type' unless EVENT_TYPES.include?(type.to_s)
+
+      @event_hash.merge!(
+        type: type,
+      )
+
+      self
+    end
+
     def with_request_details(rack_request)
       @event_hash.merge!(
         request_uuid: rack_request.uuid,
-        request_path: rack_request.path,
+        request_user_agent: rack_request.method,
         request_method: rack_request.method,
+        request_path: rack_request.path,
+        request_query: rack_request.query_string.presence || nil,
```

Comment:
Doesn't `rack_request.query_string.presence` return nil if it's not `present?`? Also, don't we risk leaking personal info by doing this? Should we be whitelisting the query params here?

```suggestion
        request_query: rack_request.query_string.presence,
```

Reply:
Ah yes, good point about `presence`. I wondered about that — TVS don't. In principle we never send  PII in query params because it also goes straight into Logit. I wonder if it might be more worthwhile to actually break these down tho — eg for analytics of filters etc?

Reply:
We could create a repeating param-value record set like the `data` one, kind of like:

```
request_query: [
  {
    param: "page",
    value: "1",
  },
  {
    param: "page_count",
    value: "4",
  },
]
```

Reply:
Yes — I think that would be nice! Will add